include/mouse.h

```diff
@@ -21,21 +22,95 @@
 
 #include "dosbox.h"
 
-void Mouse_ShowCursor(void);
-void Mouse_HideCursor(void);
+// ***************************************************************************
+// Notifications from external subsystems - all should go via these methods
+// ***************************************************************************
 
-bool Mouse_SetPS2State(bool use);
+void MOUSE_EventMoved(const int16_t x_rel, const int16_t y_rel, const uint16_t x_abs,
+                      const uint16_t y_abs, const bool is_captured);
+void MOUSE_EventPressed(const uint8_t idx);
+void MOUSE_EventReleased(const uint8_t idx);
+void MOUSE_EventWheel(const int16_t w_rel);
 
-void Mouse_ChangePS2Callback(uint16_t pseg, uint16_t pofs);
+void MOUSE_SetSensitivity(const int32_t sensitivity_x, const int32_t sensitivity_y);
+void MOUSE_NewScreenParams(const uint16_t clip_x, const uint16_t clip_y,
+                           const uint16_t res_x, const uint16_t res_y,
+                           const bool fullscreen, const uint16_t x_abs,
+                           const uint16_t y_abs);
+void MOUSE_NotifyMovedFake(); // for VMware protocol support
 
+// ***************************************************************************
+// Common structures, please only update via functions above
+// ***************************************************************************
 
-void Mouse_CursorMoved(float xrel,float yrel,float x,float y,bool emulate);
-void Mouse_CursorSet(float x,float y);
-void Mouse_ButtonPressed(uint8_t button);
-void Mouse_ButtonReleased(uint8_t button);
+class MouseInfoConfig {
+public:
+	float sensitivity_x; // sensitivity, might depend on the GUI/GFX
+	float sensitivity_y; // for scaling all relative mouse movements
 
-void Mouse_AutoLock(bool enable);
-void Mouse_BeforeNewVideoMode();
-void Mouse_AfterNewVideoMode(bool setmode);
+	MouseInfoConfig();
+};
+
+class MouseInfoVideo {
+public:
+	bool fullscreen;
+	uint16_t res_x; // resolution to which guest image is scaled, excluding
+	                // black borders
+	uint16_t res_y;
+	uint16_t clip_x; // clipping = size of black border (one side)
+	uint16_t clip_y; // clipping value - size of black border (one side)
+
+	MouseInfoVideo();
+};
+
+extern MouseInfoConfig mouse_config;
+extern MouseInfoVideo  mouse_video;
+extern bool            mouse_vmware; // true = VMware driver took over the mouse
+
+// ***************************************************************************
+// Serial mouse
+// ***************************************************************************
+
+class CSerialMouse;
+
+void MOUSESERIAL_RegisterListener(CSerialMouse &listener);
+void MOUSESERIAL_UnRegisterListener(CSerialMouse &listener);
```

Comment:
Can be const?

Reply:
Unfortunately, no. We sent mouse movement and key presses information to CSerialMouse objects, and they cause actions which do change object internal states.